semana25/rodada-cases-semana25/casePokemonGo/src/endpoints/searchSpecific.ts

```diff
@@ -0,0 +1,39 @@
+import { Request, Response } from "express";
+import { connection } from "../data/connection";
+
+export const searchSpecific = async(req:Request, res:Response) => {
+    try {
+        const {type1, type2} = req.query 
+
+
+        if(!type1 || !type2 ){
+            throw new Error("Insira as informações necessárias!")
+        }
+
+        const [checkType1] = await connection("pokemonGo")
+            .where("type1", "LIKE", `%${type1}%`)
+
+        if(!checkType1){
+            throw new Error("Tipo de pokemon não existe!")
+        }
+
+        const [checkType2] = await connection("pokemonGo")
+            .where("type2", "LIKE", `%${type2}%`)
+
+        if(!checkType2){
+            throw new Error("Tipo de Pokemon não existe!")
+        }
+
+
+        const [result]= await connection.raw(`
+            SELECT * FROM pokemonGo
+            WHERE type1 = "${type1}" AND type2 = "${type2}";
+        `);
```

Comment:
A lógica funciona, mas ela pode ser refatorada para uma única query de busca olhando para o tipo1 e tb para o tipo2 ao mesmo tempo utilizando o .andWhere():
```suggestion
        const result = await connection("nome-da-tabela")
              .where("type1", "LIKE", `%${type1}%`)
              .andWhere("type2", "LIKE", `%${type2}%`)
        
        if(!result.length){
            throw new Error("Tipo de Pokemon não existe!")
        }
        
        res.send(result)
```